src/device/device.cpp

```diff
@@ -41,13 +41,26 @@ namespace hw {
     /*  SETUP                                                                  */
     /* ======================================================================= */
 
-    static std::unique_ptr<device_registry> registry;
+    static device_registry *get_device_registry(bool clear = false){
+      static device_registry *registry = new device_registry();
```

Comment:
I think `static std::unique_ptr<device_registry> registry{new device_registry};` (inside of `get_device_registry`) is equivalent. AFAIK, compilers typically use `atexit` to destruct local static objects - the standard mandates that local static destructors and `atexit` functions be sequenced in reverse order.